skynet/constants.py

```diff
@@ -1,113 +1,115 @@
-#!/usr/bin/python
 import msgspec
+
+from enum import Enum
 from typing import Literal
 
 VERSION = '0.1a12'
 
 DOCKER_RUNTIME_CUDA = 'skynet:runtime-cuda'
 
 
-class Size(msgspec.Struct):
-    w: int
-    h: int
-
-
 class ModelDesc(msgspec.Struct):
-    short: str
-    mem: float
-    size: Size
-    tags: list[Literal['txt2img', 'img2img', 'inpaint']]
+    short: str  # short unique name
+    mem: float  # recomended mem
+    attrs: dict  # additional mode specific attrs
```

Comment:
hmm maybe a better field name then? like say `mode_vars` or `mode_params`?

Reply:
also maybe longer term it's' wise for us to require model backends to declare `msgspec.Struct`s for these themselves?

Reply:
That field is meant to be extremely generic, keys on that field can change from model to model depending on what the model does, for example for images we store preferred size information (some models operate better at certain img sizes), but in the case of text models that is not needed, but we might need other types of information like max context window, etc.